todo_app/data/mongo_items.py

```diff
@@ -10,42 +10,45 @@
 from todo_app.data.mongoList import MongoList
 import todo_app.data.mongo_constants as constants
 import iso8601
-
+import logging as logger
 class Mongo_service(object):
     mongo_lists = {}
      
     def get_mongo_params(self):
-        return { 'user_name': os.getenv('USER_NAME'), 
-                'password': os.getenv('PASSWORD'),
-                'mongo_url': os.getenv('MONGO_URL'),
-                'default_db': os.getenv('DEFAULT_DATABASE'),
-                'mongo_prefix': os.getenv('MONGO_PREFIX'),
-                'mongo_config': os.getenv('MONGO_CONFIG')}
+        return { 'mongo_connection_string': os.getenv('MONGO_CONNECTION_STRING'),
+                 'default_db': os.getenv('DEFAULT_DATABASE')
+        }
 
     def initiate(self):
         self.mongo_lists = {}
         self.init_mongo_db()
 
     def init_mongo_db(self):
         mongo_config = self.get_mongo_params()
-        mongo_user = mongo_config ['user_name']
-        mongo_password = mongo_config ['password']
-        mongo_url = mongo_config ['mongo_url']
-        mongo_db = mongo_config ['default_db']
-        mongo_prefix = mongo_config ['mongo_prefix']
-        mongo_cfg = mongo_config ['mongo_config']
-        mongo_connection_string = f"{mongo_prefix}{mongo_user}:{mongo_password}@{mongo_user}.{mongo_url}/{mongo_db}{mongo_cfg}"
-        mongo_client = pymongo.MongoClient(mongo_connection_string)
-        self.client = mongo_client
-        self.db = mongo_client[f'{mongo_db}']
+        mongo_connection_string = mongo_config ['mongo_connection_string']
+        mongo_db = mongo_config [ 'default_db']
+        logger.debug("Creating mongo client")
+
+        try:
+          mongo_client = pymongo.MongoClient(mongo_connection_string)
+          self.client = mongo_client
+          self.db = mongo_client[f'{mongo_db}']
+        except:
+          logger.exception("Error initiating the Mongo client!")
```

Comment:
This pattern of
```python
try:
  ...
except:
  logger.exception('Bad things happened')
```
can be a bit dangerous. Here `init_mongo_db` will return as normal and execution will carry on with an incorrectly initialised database, most likely causing another exception or bad behaviour to happen later, which can be harder to debug.

For this kind of thing where you haven't really handled the exception you should re-raise it
```python
try:
  ...
except:
  logger.exception('Bad things happened')
  raise
```
which is short for
```python
try:
  ...
except Exception as e:
  logger.exception('Bad things happened')
  raise e
```

Reply:
Will change in my module 14 which is nearing its completion.